Makefile

```diff
@@ -207,13 +207,18 @@ sync-docs:
 test: test-unit
 test-all: test-unit test-ledger-mock test-race test-cover
 
+# only compiles the tests without
+# note: go test -c doesn't support multiple packages yet (https://github.com/golang/go/issues/15513)
+test-build-check:
```

Comment:
`make test-all` runs all tests and leverages `go cache` to prevent tests from being unnecessarily re-executed. So what value does this bring to the party exactly?

Reply:
Much shorter time --> improves DevUX.

Reply:
How does this shorten test execution time? This command just skips lots of tests, e.g. ledger related ones.

Reply:
`test-build-check` command doesn't execute any test. It's helpful to do a quick check without taking a break.